src/Twig/Extension/PageExtension.php

```diff
@@ -218,4 +221,30 @@ public function controller(string $controller, array $attributes = [], array $qu
 
         return HttpKernelExtension::controller($controller, $attributes, $query);
     }
+    
+    /**
+     * @param PageInterface $page
+     * @param array $parameters
+     * @return string
+     */
+    public function pageUrl(PageInterface $page, array $parameters = []) : string
+    {
+        $parameters = array_merge($parameters, [
+            RouteObjectInterface::ROUTE_OBJECT => $page
+        ]);
+        return $this->router->generate(RouteObjectInterface::OBJECT_BASED_ROUTE_NAME, $parameters, RouterInterface::ABSOLUTE_URL);
+    }
+
+    /**
+     * @param PageInterface $page
+     * @param array $parameters
+     * @return string
+     */
+    public function pagePath(PageInterface $page, array $parameters = []) : string
+    {
+        $parameters = array_merge($parameters, [
+            RouteObjectInterface::ROUTE_OBJECT => $page
+        ]);
+        return $this->router->generate(RouteObjectInterface::OBJECT_BASED_ROUTE_NAME, $parameters, RouterInterface::ABSOLUTE_PATH);
+    }    
```

Comment:
The implementation I had in mind was:
```
    /**
     * @param string|PageInterface $name
     * @param array<mixed>         $parameters
     */
    public function getPagePath(string $name, array $parameters = [], bool $relative = false): string
    {
        return $this->cmsPageRouter->generate($name, $parameters, $relative ? UrlGeneratorInterface::RELATIVE_PATH : UrlGeneratorInterface::ABSOLUTE_PATH);
    }

    /**
     * @param string|PageInterface $name
     * @param array<mixed>         $parameters
     */
    public function getPageUrl(string $name, array $parameters = [], bool $schemeRelative = false): string
    {
        return $this->cmsPageRouter->generate($name, $parameters, $schemeRelative ? UrlGeneratorInterface::NETWORK_PATH : UrlGeneratorInterface::ABSOLUTE_URL);
    }
```

This way, you get the same behavior than before, and no need for OBJECT_BASED_ROUTE_NAME